dogstatsd/src/dogstatsd.rs

```diff
@@ -184,6 +187,34 @@ single_machine_performance.rouster.metrics_max_timestamp_latency:1376.90870216|d
         assert_value(&locked_aggregator, "metric123", 99_123.0, "");
     }
 
+    #[tokio::test]
+    #[traced_test]
+    #[cfg_attr(miri, ignore)]
+    async fn test_dogstatsd_filter_service_check() {
+        let locked_aggregator = setup_dogstatsd("_sc|servicecheck|0").await;
+        let aggregator = locked_aggregator.lock().expect("lock poisoned");
+        let parsed_metrics = aggregator.to_series();
+
+        assert!(!logs_contain("Failed to parse metric"));
+        assert_eq!(parsed_metrics.len(), 0);
+
+        drop(aggregator);
```

Comment:
do we need these explicit drop calls? It should happen automatically at the end of the block I think

Reply:
I was following the pattern in the other unit tests but if you think it's not needed then I can remove it.